ipa-core/src/net/server/handlers/mod.rs

```diff
@@ -6,12 +6,14 @@ use axum::Router;
 use crate::net::{http_serde, transport::MpcHttpTransport, ShardHttpTransport};
 
 pub fn mpc_router(transport: MpcHttpTransport) -> Router {
-    echo::router().nest(
-        http_serde::query::BASE_AXUM_PATH,
-        Router::new()
-            .merge(query::query_router(transport.clone()))
-            .merge(query::h2h_router(transport)),
-    )
+    echo::router()
+        .nest(
+            http_serde::query::BASE_AXUM_PATH,
+            Router::new()
+                .merge(query::query_router(transport.clone()))
+                .merge(query::h2h_router(transport.clone())),
+        )
+        .merge(query::metric_router(transport))
```

Comment:
Nit: Does this handler belong inside the query module? You can perhaps park it alongside `echo`